### ISSUE TITLE
CPU usage high and status bar flickers when works with VSCode Vim

### ISSUE THREAD

**Issue by houkanshan** [Jason-Rev/vscode-spell-checker#153]
Hi, thank you for your extremely useful extension. This problem happened when I just upgrade several things including VSCode@1.18.0, VSCodeVim@0.10.4

**ENV**:
macOS Sierra 10.12.6
VSCode@1.18.0
VSCodeVim@0.10.4
Code Spell Checker@1.4.12

![2017-11-15 11_25_31](https://user-images.githubusercontent.com/657962/32817266-c11f25ae-c9f7-11e7-940f-428d9b015f39.gif)

In the GIF, the status bar is flickering between these 2 states

![image](https://user-images.githubusercontent.com/657962/32817421-7d8b1ad6-c9f8-11e7-84de-e1f82176e8f0.png)

![image](https://user-images.githubusercontent.com/657962/32817391-5545588e-c9f8-11e7-876a-f5a4f01fd4f5.png)

Also, the count of spell errors in the red circle will occasionally change from 25 to 39.

It will happen when I disabled all the extensions except Code Spell Checker & VSCodeVim, and the VSCodeVim has enabled `statusBarColorControl` like this:

```js
    "vim.statusBarColorControl": true,
    "vim.statusBarColors": {
      "normal": "#005f5f",
      "insert": "#5f0000",
      "visual": "#5f00af",
      "visualline": "#005f87",
      "visualblock": "#86592d",
      "replace": "#000000"
    },
```

This problem can be solved temporarily by re-save the `settings.json`. Or just disabled the `vim.statusBarColorControl` feature. 

Since it could be a bug of VSCode or VSCodeVim, and the status bar color is unnecessary to me, I just put this issue here in case you or others will find it useful.

**Comment by Jason3S** [comment]
Wow, thank you for such a great and explicit example!

I'll see if I can figure out how the two extensions are conflicting.

Can you try:

```
"vim.statusBarColorControl": true,
"cSpell.showStatus": false,
```

and check to see if the High CPU still happens?

**Comment by houkanshan** [comment]
Hi, thanks for replying.

After setting `"cSpell.showStatus": false`, the clock icon disappeared. But the count of spell errors is still flickering, and CPU is still high.

![2017-11-16 11_30_03](https://user-images.githubusercontent.com/657962/32872433-9fade282-cac1-11e7-9d89-4a1792f95461.gif)

And I figure out what the flickering numbers are – in the GIF above, 5 is the warning count of the current file, 19 is the count of all the open files (= 14 + 5).

**Comment by Jason3S** [comment]
Wow, it is acting like something is changing `settings.json` in the background.

Whenever `settings.json` is changed, the spell checker has to reload all the settings and start the spell checking process from scratch. The **clock** is displayed while that is happening.

**Comment by houkanshan** [comment]
You are right, every time I change the Vim editing mode, the update time of `setting.json` will change. I think VSCode Vim is changing it. 

Also, it seems Spell Checker is not the only extension that has such conflict, I found this issue: https://github.com/VSCodeVim/Vim/issues/2124

It should be a problem of VSCode Vim, I'm closing this issue, thank you for your help!

**Comment by Jason3S** [comment]
Thank you for reporting this.

I am guessing that the Vim extension is changing the `settings.json` file (possibly through the VS Code configuration api `workspace.getConfiguration().update()`) to change the status-bar color.

If VS Code does not correctly filter out the changes, it will cause the spell checker to re-load all configuration and re-check ALL open files. I think this is fixed in VS Code 1.19: https://github.com/Microsoft/vscode/issues/36475

**Comment by houkanshan** [comment]
Great, waiting for the stable build of 1.19

You are really helpful. 🥇
